app/Http/Controllers/EventAPIController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Swagger\EventAPISwagger;
+use App\Http\Requests\ListEventRequest;
+use App\Http\Resources\EventResource;
+use App\Services\Contracts\EventServiceContract;
+use EscolaLms\Core\Dtos\OrderDto;
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+use EscolaLms\StationaryEvents\Enum\ConstantEnum;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Pagination\LengthAwarePaginator;
+
+class EventAPIController extends EscolaLmsBaseController implements EventAPISwagger
+{
+    private EventServiceContract $eventService;
+
+    public function __construct(EventServiceContract $eventService)
+    {
+        $this->eventService = $eventService;
+    }
+
+    public function index(ListEventRequest $request): JsonResponse
+    {
+        $orderDto = OrderDto::instantiateFromRequest($request);
+        $page = LengthAwarePaginator::resolveCurrentPage();
+        $perPage = $request->get('per_page') ?? ConstantEnum::PER_PAGE;
+
+        $events = $this->eventService->getEventsList($orderDto);
+        $result = new LengthAwarePaginator($events->forPage($page, $perPage), $events->count(), $perPage, $page, [
```

Comment:
This retrieves all webinars and events and then extracts part of the returned collection and builds a paginator from it?

I think what we need to do instead, is to create a database view that represents searchable data in unified format, query that view, and return stuff from it using polymorphic relation.

There is a library for migrating views https://github.com/staudenmeir/laravel-migration-views

And we need a view that joins data from both tables using UNION
so something like
```
SELECT
webinars.id as event_id,
"EscolaLms\Webinars\Models\Webinar" as event_type,
webinars.active_from as start_date,
webinars.active_to as end_date,
UNION
SELECT 
stationary.id as event_id,
"EscolaLms\StationaryEvents\Models\StationaryEvent" as event_class,
stationary.start_date as start_date,
stationary.end_date as end_date,
```
etc, then query and paginate this view using Model (I'm pretty sure that we can just create a model that has $table='view_name' and it should work as if it was using a table) and that Model should have method
```
event(): MorphTo {
    return $this->morphTo()
} 
```
and then we need to just return JsonResource that extracts that morphed event data

I'm not sure if we ever did something like this but it's worth a try XD